test/test_hpytype.py

```diff
@@ -221,6 +221,338 @@ def test_HPyDef_Member(self):
+            HPyDef_MEMBER(Foo_STRING_member, "STRING_member", HPyMember_STRING, offsetof(FooObject, STRING_member))
+            HPyDef_MEMBER(Foo_CHAR_member, "CHAR_member", HPyMember_CHAR, offsetof(FooObject, CHAR_member))
+            HPyDef_MEMBER(Foo_ISTRING_member, "ISTRING_member", HPyMember_STRING_INPLACE, offsetof(FooObject, ISTRING_member))
+            HPyDef_MEMBER(Foo_BOOL_member, "BOOL_member", HPyMember_BOOL, offsetof(FooObject, BOOL_member))
+            HPyDef_MEMBER(Foo_NONE_member, "NONE_member", HPyMember_NONE, offsetof(FooObject, FLOAT_member))
+
+            static HPyDef *Foo_defines[] = {
+                    &Foo_new,
+                    &Foo_FLOAT_member,
+                    &Foo_DOUBLE_member,
+                    &Foo_STRING_member,
+                    &Foo_CHAR_member,
+                    &Foo_ISTRING_member,
+                    &Foo_BOOL_member,
+                    &Foo_NONE_member,
+                    NULL
+            };
+
+            static HPyType_Spec Foo_spec = {
+                .name = "mytest.Foo",
+                .basicsize = sizeof(FooObject),
+                .defines = Foo_defines
+            };
+
+            @EXPORT_TYPE("Foo", Foo_spec)
+            @INIT
+            """)
+        foo = mod.Foo()
+        assert foo.FLOAT_member == 1.
+        foo.FLOAT_member = 0.1
+        assert foo.FLOAT_member != 0.1
+        assert abs(foo.FLOAT_member - 0.1) < 1e-8
+        with pytest.raises(TypeError):
+            del foo.FLOAT_member
+
+        assert foo.DOUBLE_member == 1.
+        foo.DOUBLE_member = 0.1
+        assert foo.DOUBLE_member == 0.1  # exactly
+        with pytest.raises(TypeError):
+            del foo.DOUBLE_member
+
+        assert foo.STRING_member == "Hello"
+        with pytest.raises(TypeError):
+            foo.STRING_member = "world"
+        with pytest.raises(TypeError):
+            del foo.STRING_member
+
+        assert foo.CHAR_member == 'A'
+        foo.CHAR_member = 'B'
+        assert foo.CHAR_member == 'B'
+        with pytest.raises(TypeError):
+            foo.CHAR_member = 'ABC'
+        with pytest.raises(TypeError):
+            del foo.CHAR_member
+
+        assert foo.ISTRING_member == "Hello"
+        with pytest.raises(TypeError):
+            foo.ISTRING_member = "world"
+        with pytest.raises(TypeError):
+            del foo.ISTRING_member
+
+        assert foo.BOOL_member is False
+        foo.BOOL_member = True
+        assert foo.BOOL_member is True
+        with pytest.raises(TypeError):
+            foo.BOOL_member = 1
+        with pytest.raises(TypeError):
+            del foo.BOOL_member
+
+        assert foo.NONE_member is None
+        with pytest.raises((SystemError, TypeError)):  # CPython quirk/bug
+            foo.NONE_member = None
+        with pytest.raises(TypeError):
+            del foo.NONE_member
+
+    def test_HPyDef_Member_readonly(self):
```

Comment:
I guess this test should be named `test_HPyDef_Member_readonly_others` for consistency?